tests/appsec/test_metastruct.py

```diff
@@ -0,0 +1,94 @@
+# Unless explicitly stated otherwise all files in this repository are licensed under the the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2021 Datadog, Inc.
+
+from utils import weblog, context, interfaces, bug, missing_feature, scenarios, features
+
+
+@features.security_events_metastruct
+class Test_SecurityEvents_Metastruct:
+    """Tests to verify that security events are correctly set in meta struct when the agent supports it."""
+
+    def setup_appsec_event_use_metastruct(self):
+        self.r = weblog.get("/", headers={"User-Agent": "Arachni/v1"})
+
+    def test_appsec_event_use_metastruct(self):
+        spans = [s for _, s in interfaces.library.get_root_spans(request=self.r)]
+        assert spans
+
+        for span in spans:
+            meta = span.get("meta", {})
+            meta_struct = span.get("meta_struct", {})
+            assert meta["appsec.event"] == "true"
+            assert "_dd.appsec.json" not in meta
+            assert "appsec" in meta_struct
+
+            # The event is not null
+            assert meta_struct.get("appsec", {}) not in [None, {}]
+
+            # There is at least one rule triggered
+            assert len(meta_struct["appsec"].get("triggers", [])) > 0
+
+    def setup_iast_event_use_metastruct(self):
+        # Using this vulnerability because that's one that is implemented in all tracers
+        self.r = weblog.post("/iast/cmdi/test_insecure", data={"cmd": "echo 'metastruct'"})
```

Comment:
I'm not sure if it's still applies today, but tests used to run in the same instance of a class, and using `self.r` in two setups would give unpredictable results (one instance or the other).

Reply:
Tests has been splitted into multiple class so it is more easier to implement and won't create new easy wins.